### ISSUE TITLE
[$250] Can't view expense details for card transactions imported on custom feed

### ISSUE THREAD

**Issue by joekaufmanexpensify** [Expensify/App#54393]
# Problem
As shown [here](https://expensify.slack.com/archives/C06ML6X0W9L/p1734634118861409?thread_ts=1734560659.771669&cid=C06ML6X0W9L), if you import a card from a custom card feed that has transactions, we show the transactions imported in the report preview of the workspace chat. However, if you click into the report, there is no list of expenses shown. Further, we show the individual expenses on the search page, but if you click on an expense, you see the "hmmm not here page". 

https://github.com/user-attachments/assets/3e364b08-b9e5-4c28-bf2f-f0d38de4666e

# Solution 
You should be to view the details of any card-imported transaction on both a custom or direct feed. on both the expense report in the workspace chat or on the search page. 

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021870477160211667610</li>
        <li>Upwork Job ID: 1870477160211667610</li>
        <li>Last Price Increase: 2024-12-28</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

**Comment by joekaufmanexpensify** [comment]
Note: this is working fine with direct feeds. It is only custom feeds having this issue.

**Comment by DylanDylann** [comment]
I would love to help on this issue as C+

**Comment by mountiny** [comment]
@DylanDylann cam you test this out and try to triage where the problem is?

**Comment by DylanDylann** [comment]
@mountiny Yes. I am asking for support on [Slack](https://expensify.slack.com/archives/C06ML6X0W9L/p1734925561533519?thread_ts=1734560659.771669&cid=C06ML6X0W9L)

**Comment by DylanDylann** [comment]
I already imported the Onyx file from @joekaufmanexpensify and see that we have some problems with data (need to check on BE to understand why the BE returned  this data)

[onyx-state-jkaufman+customfeedstest@expensifail.com (1).txt](https://github.com/user-attachments/files/18240946/onyx-state-jkaufman%2Bcustomfeedstest%40expensifail.com.1.txt)


In this onyx data, we had 26 transactions (that has transaction.reportID = 4610253791484989)

<img width="416" alt="Screenshot 2024-12-24 at 23 35 56" src="https://github.com/user-attachments/assets/38cbb63e-b771-4edc-9454-9f7ea9b2ecd9" />

these transactions will be matched to the iouReport (that has ID 4610253791484989). It explains why we see 26 transactions in the report preivew

<img width="594" alt="Screenshot 2024-12-24 at 23 37 38" src="https://github.com/user-attachments/assets/89bc1ed0-0707-4fcb-9238-5934b4702c4f" />

But if we click on iou report we don't see any comment

<img width="975" alt="Screenshot 2024-12-24 at 23 38 57" src="https://github.com/user-attachments/assets/e1ad3a4a-2c41-410d-95df-c1db8ace12d7" />


Because reportActions_4610253791484989 only has 3 reportActions it is so confusing to me because we have 26 transactions on this expense

In 3 report actions:
- CREATED action
- IOU action (it is hidden because it is marked as deleted)
- DELETEDTRANSACTION action (it is hidden because the FE haven't supported it)

**Comment by mountiny** [comment]
Havent got around to this one yet

**Comment by mountiny** [comment]
PR is up https://github.com/Expensify/Auth/pull/13674 this seems to be because we are unreliably picking the policy o which to add these new scraped transactions

**Comment by mountiny** [comment]
One more PR was required and its merged awaiting deployment

**Comment by mountiny** [comment]
Confirmed with Jason this is fixed
